scripts/combine-ui-test-results.js

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env node
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+'use strict';
+const fs = require('fs');
+
+const NUM_PARTITIONS = 4;
+
+function combineResults() {
+  const results = [];
+  let duration = 0;
+  let aggregateSummary = { total: 0, passed: 0, failed: 0 };
+
+  for (let i = 1; i <= NUM_PARTITIONS; i++) {
+    try {
+      const data = JSON.parse(
+        fs.readFileSync(`../test-results/test-results-${i}/test-results.json`).toString()
+      );
```

Comment:
Will there be anything else in the test-results directory other than subdirectories for each partition?

I wonder if you can walk the dir tree instead of relying on synchronizing `NUM_PARTITIONS` and the GHA matrix config.